src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -207,66 +324,171 @@ def __exit__(self, exc_type, exc_val, exc_tb):
+                self._completed_idx[line_number] = format_current_process(process_name, pid, line_number, True)
                 if not task_queue.empty():
-                    healthy_ensured_process.end()
-                    healthy_ensured_process.start_new(task_queue)
+                    if self._use_fork:
+                        self._control_signal_queue.put((pid, index, "restart"))
+                        process_info = self._get_process_info(
+                            input_queue=input_queue,
+                            output_queue=output_queue
+                        )
+                    else:
+                        healthy_ensured_process.end()
+                        healthy_ensured_process.start_new()
+                        process_info = self._get_process_info(healthy_ensured_process=healthy_ensured_process)
 
             self._processing_idx.pop(line_number)
+
             log_progress(
                 run_start_time=run_start_time,
                 logger=bulk_logger,
                 count=len(result_list),
                 total_count=self._nlines,
             )
 
+    def _get_process_info(self, input_queue=None, output_queue=None, healthy_ensured_process=None):
+        # Using fork
+        if input_queue and output_queue:
+            index, pid, process_name = input_queue.get()
+            process_info = (index, pid, process_name, input_queue, output_queue)
+        elif healthy_ensured_process is not None:
+            process_info = (
+                healthy_ensured_process,
+                healthy_ensured_process.process.pid,
+                healthy_ensured_process.process.name,
+                healthy_ensured_process.input_queue,
+                healthy_ensured_process.output_queue
+            )
+        return process_info
+
+    def _process_message(self, message, result_list):
+        if isinstance(message, LineResult):
+            message = self._process_multimedia(message)
+            result_list.append(message)
+            return True
+        elif isinstance(message, FlowRunInfo):
+            self._storage.persist_flow_run(message)
+        elif isinstance(message, NodeRunInfo):
+            self._storage.persist_node_run(message)
+
+        return False
+
+    def handle_line_timeout(self, line_number, timeout_time, inputs, run_id, start_time, result_list):
+        bulk_logger.warning(f"Line {line_number} timeout after {timeout_time} seconds.")
+        ex = LineExecutionTimeoutError(line_number, timeout_time)
+        result = self._generate_line_result_for_exception(
+            inputs, run_id, line_number, self._flow_id, start_time, ex
+        )
+        result_list.append(result)
+
+    def _timeout_process_wrapper(
+            self,
+            run_start_time: datetime,
+            task_queue: Queue,
+            timeout_time,
+            result_list,
+            input_queue=None,
+            output_queue=None
+    ):
+        if self._use_fork:
```

Comment:
It may not be a good practice to have a lot of `self._user_fork` in different places.
We may have two choices:
1. If we can have most logic common, we may try put the difference only when initializing the processes, then use the same interface to consume input data.
2. If most logic cannot be shared, we may have two different classes to handle the two scenarios.

Personally I prefer 1, but it seems that the current implementations have a lot of fork specific logics.

Reply:
I think we'd be better revisit the interface design before do the implementation.

Reply:
Just a simple proposal according to this PR:
```
class AbstractProcessManager:
    def create_process(idx):
        pass
    def restart_process(idx):
        pass
    
For SpawnProcessManager, here we just simply start and stop processes.
For ForkProcessManager, here we will send a signal to the spawned main process for creation, and send another signal to the spawned process for stop and restart.
```

Reply:
We may add more interfaces to handle the task consumption and timeout monitoring.

Reply:
fixed